actions/HistoryActions.js

```diff
@@ -0,0 +1,186 @@
+import * as Wallet from '../app/services/wallet'
+
+export const ACTIONS = {
+  LOAD_CHANNELS: 'channels/load',
+  LOAD_INVOICES: 'invoices/load',
+  LOAD_MORE_INVOICES: 'invoices/loadMore',
+  LOAD_PEERS: 'peers/load',
+  LOAD_PAYMENTS: 'payments/load',
+  LOAD_MORE_PAYMENTS: 'payments/loadMore',
+  LOAD_TRANSACTIONS: 'transactions/load',
+  LOAD_MORE_TRANSACTIONS: 'transactions/loadMore',
+  LOAD_RECENT_TRANSACTIONS: 'recentTransactions/load',
+}
+/**
+ * Fetches the Node's info
+ * @returns {import('redux-thunk').ThunkAction<Promise<Wallet.Channel[]>, {}, {}, import('redux').AnyAction>}
+ */
+export const fetchChannels = () => async dispatch => {
```

Comment:
Pagination is not yet implemented in the API, will have to implement it there first before adding it here

Reply:
I think this is an anti-pattern. All of the state handling should occur inside the component instead of the action, which makes it a lot easier to maintain too. You can always access the current page from the HistoryReducer in that component.